librouteros/query.py

```diff
@@ -71,3 +71,24 @@ def Or(left: QueryGen, right: QueryGen, *rest: QueryGen) -> QueryGen:
     yield from chain.from_iterable(rest)
     yield "?#|"
     yield from ("?#|",) * len(rest)
+
+
+class AsyncQuery:
+    def __init__(self, path, keys: typing.Sequence[Key], api):
+        self.path = path
+        self.keys = keys
+        self.api = api
+        self.query: typing.Tuple[str, ...] = tuple()
+
+    def where(self, *args: str):
+        self.query = tuple(chain.from_iterable(args))
+        return self
+
+    def __aiter__(self):
+        keys = ",".join(str(key) for key in self.keys)
+        keys = f"=.proplist={keys}"
+        cmd = str(self.path.join("print"))
+        return self.api.rawCmd(cmd, keys, *self.query)
```

Comment:
```suggestion
        words = tuple(self.query)
        if len(self.keys) > 0:
            keys = ",".join(str(key) for key in self.keys)
            keys = f"=.proplist={keys}"
            words = (keys,) + words
        return iter(self.api.rawCmd(cmd, *words))
```
From https://github.com/luqasz/librouteros/commit/1ee5479df767e59923f35e2c2ed124f17d0bf186